projects/odf_transform/odf_transform/odf_parser/odf_parser.py

```diff
@@ -165,68 +171,119 @@ def _convert_to_number(value):
+        is_q_flag = var.startswith("Q") and var[:1] in metadata.keys()
+        is_qqqq_flag = odf_var_name['name'] == 'QQQQ'
+        is_general_flag = odf_var_name['name'] in ['QCFF', 'FFFF']
+
+        is_flag_column = is_q_flag or is_qqqq_flag or is_general_flag
+
+        # Find related variable
+        if is_qqqq_flag:
+            # MLI FLAG QQQQ should apply to previous variable
+            related_variable = previous_key
+        if is_q_flag:
+            # Q  Format is usually Q+[PCODE] of the associated variable
+            related_variable = var[1:]
+
+        # Set previous key for the next iteration
+        previous_key = var
+
+        # If the variable isn't a flag variable, go to the next iteration
+        if not is_flag_column:
+            continue
+
+        # If flag is specific to a variable, try to confirm if the odf name of the related variable match the
+        # flag name. If yes, add this flag variable to the ancillary_attribute of the related variable
+        if related_variable:
+            # FIRST MAKE SURE THE RELATED VARIABLE IS THE RIGHT ONE
+            # Make sure that the related variable exist
+            if related_variable not in metadata:
+                raise KeyError(
+                    '{0} flag is referring to {1} which is not available as variable'.format(
+                        var, related_variable),
+                    )
+
+            # Try to see if the related_variable and flag have matching name or code in odf
+            related_variable_name = re.sub(r"quality\sflag.*:\s*|quality flag of ", '',
+                                           att['original_NAME'], 1, re.IGNORECASE)
+            # Flag name do not match either variable name or code, give a warning.
+            if related_variable_name not in metadata[related_variable].get('original_NAME') and \
+                    related_variable_name not in metadata[related_variable].get('original_CODE'):
+                warnings.warn(
+                    '{0}[{1}] flag was matched to the variable {2} but ODF attributes {3} do not match'.format(
+                        var,
+                        att['original_NAME'],
+                        related_variable,
+                        {key: metadata[related_variable].get(key) for key in ['original_NAME', 'original_CODE']}
+                    ),
+                    UserWarning
+                )
+
+            # Standardize long name attribute
+            if 'name' in metadata[related_variable]:
+                att['long_name'] = flag_long_name_prefix + metadata[related_variable]['name']
+            else:
+                att['long_name'] = flag_long_name_prefix + related_variable
+
+            # Add ancillary_variables attribute
+            if 'ancillary_variables' not in metadata[related_variable]:
+                metadata[related_variable]['ancillary_variables'] = var
+            elif 'ancillary_variables' in metadata[related_variable]:
```

Comment:
dont need `elif` here, can just be else, also best to do the positive condition first- `if a, else` instead of `if not a, else`

Reply:
those were inverted